`encoding_struct!` needs explicit `serde_json` dependency

To use `encoding_struct!` macro I have to add `serde_json` dependency in every project, but we can reexport that crate and use it transparently.

```rust
error[E0433]: failed to resolve. Maybe a missing `extern crate serde_json;`?
  --> src/main.rs:15:1
   |
15 | / encoding_struct! {
16 | |     struct Wallet {
17 | |         const SIZE = 48;
18 | |
...  |
22 | |     }
23 | | }
   | |_^ Maybe a missing `extern crate serde_json;`?
   |
   = note: this error originates in a macro outside of the current crate
```

--- a/exonum/src/encoding/spec.rs
+++ b/exonum/src/encoding/spec.rs
@@ -183,7 +183,7 @@ macro_rules! encoding_struct {
             }
         }
         impl $crate::encoding::serialize::json::ExonumJsonDeserialize for $name {
-            fn deserialize(value: &::serde_json::Value) -> Result<Self, Box<::std::error::Error>> {
+            fn deserialize(value: &$crate::encoding::serialize::json::reexport::Value) -> Result<Self, Box<::std::error::Error>> {
                 let to = $body as $crate::encoding::Offset;
                 let from = 0;
                 use $crate::encoding::serialize::json::ExonumJson;

